spot_ros2_control/hardware/include/spot_ros2_control/spot_hardware_interface.hpp

```diff
@@ -0,0 +1,131 @@
+// File modified. Modifications Copyright (c) 2024 Boston Dynamics AI Institute LLC.
+// All rights reserved.
+
+// --------------------------------------------------------------
+// Copyright 2020 ros2_control Development Team
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef SPOT_ROS2_CONTROL__SPOT_HPP_
+#define SPOT_ROS2_CONTROL__SPOT_HPP_
+
+#include <functional>
+#include <memory>
+#include <string>
+#include <thread>
+#include <vector>
+
+#include "hardware_interface/handle.hpp"
+#include "hardware_interface/hardware_info.hpp"
+#include "hardware_interface/system_interface.hpp"
+#include "hardware_interface/types/hardware_interface_return_values.hpp"
+#include "rclcpp/macros.hpp"
+#include "rclcpp_lifecycle/node_interfaces/lifecycle_node_interface.hpp"
+#include "rclcpp_lifecycle/state.hpp"
+#include "spot_ros2_control/visibility_control.h"
+
+#include "bosdyn/client/lease/lease_keepalive.h"
+#include "bosdyn/client/robot_command/robot_command_client.h"
+#include "bosdyn/client/robot_command/robot_command_helpers.h"
+#include "bosdyn/client/robot_command/robot_command_streaming_client.h"
+#include "bosdyn/client/robot_state/robot_state_client.h"
+#include "bosdyn/client/robot_state/robot_state_streaming_client.h"
+#include "bosdyn/client/sdk/client_sdk.h"
+#include "bosdyn/client/service_client/client_header_handling.h"
+#include "bosdyn/client/time_sync/time_sync_helpers.h"
+#include "bosdyn/client/util/cli_util.h"
+
+using StateHandler = std::function<void(::bosdyn::api::RobotStateStreamResponse&)>;
+
+namespace spot_ros2_control {
+
+class StateStreamingHandler {
+ public:
+  void handle_state_streaming(::bosdyn::api::RobotStateStreamResponse& robot_state);
+  std::vector<float> get_position();
+  std::vector<float> get_velocity();
+  std::vector<float> get_load();
```

Comment:
Same question about consistency but with formatting?

Reply:
Do we want copies or should we add `const` functions with `const &` return types?

Reply:
Yeah they should probably return `const&`, will update